pygmt/clib/session.py

```diff
@@ -494,6 +494,97 @@ def get_default(self, name):
 
         return value.value.decode()
 
+    def get_common(self, option):
+        """
+        Inquire if a GMT common option has been set and return its current
+        value if possible.
+
+        Parameters
+        ----------
+        option : str
+            The GMT common option to check. Valid options are ``"B"``, ``"I"``,
+            ``"J"``, ``"R"``, ``"U"``, ``"V"``, ``"X"``, ``"Y"``, ``"a"``,
+            ``"b"``, ``"f"``, ``"g"``, ``"h"``, ``"i"``, ``"n"``, ``"o"``,
+            ``"p"``, ``"r"``, ``"s"``, ``"t"``, and ``":"``.
+
+        Returns
+        -------
+        value : bool, int, float, or numpy.ndarray
+            Whether the option was set or its value.
+
+            If the option was not set, return ``False``. Otherwise,
+            the return value depends on the choice of the option.
+
+            - options ``"B"``, ``"J"``, ``"U"``, ``"g"``, ``"n"``, ``"p"``,
+              and ``"s"``: return ``True``
+            - ``"I"``: 2-element array for the increments (float)
+            - ``"R"``: 4-element array for the region (float)
+            - ``"V"``: the verbose level (int)
+            - ``"X"``: the xshift (float)
+            - ``"Y"``: the yshift (float)
+            - ``"a"``: geometry of the dataset (int)
+            - ``"b"``: the option was set for input or output (int)
+            - ``"f"``: the option was set for input or output (int)
+            - ``"h"``: whether to delete existing header records (int)
+            - ``"i"``: number of input columns (int)
+            - ``"o"``: number of output columns (int)
+            - ``"r"``: registration type (int)
+            - ``"t"``: 2-element array for the transparency (float)
+            - ``":"``: the option was set for input or output (int)
+
+        Examples
+        --------
+        >>> with Session() as lib:
+        ...     lib.call_module("basemap", "-R0/10/10/15 -JX5i/2.5i -Baf -Ve")
+        ...     region = lib.get_common("R")
+        ...     projection = lib.get_common("J")
+        ...     timestamp = lib.get_common("U")
+        ...     verbose = lib.get_common("V")
+        ...     lib.call_module("plot", "-T -Xw+1i -Yh-1i")
+        ...     xshift = lib.get_common("X")  # xshift/yshift are in inches
```

Comment:
Does GMT_Get_Common return numbers in inches only? I.e. even if we do `-JX5c/2.5c`, the output value is still converted from centimeters to inches?

Reply:
> Does GMT_Get_Common return numbers in inches only? I.e. even if we do `-Xw+1c`, the output value is still converted from centimeters to inches?

Yes, I believe internally GMT stores all dimensions in inches or points, most likely because PostScript uses points as its default unit (https://en.wikipedia.org/wiki/PostScript#Units_of_length), and 72 points are exactly 1 inch, but 72/2.54 points are 1 cm.

Reply:
> > Does GMT_Get_Common return numbers in inches only? I.e. even if we do `-Xw+1c`, the output value is still converted from centimeters to inches?
> 
> Yes, I believe internally GMT stores all dimensions in inches or points, most likely because PostScript uses points as its default unit (https://en.wikipedia.org/wiki/PostScript#Units_of_length), and 72 points are exactly 1 inch, but 72/2.54 points are 1 cm.

That's also why I use `-Xw+1i` in the docstrings, which makes the doctest easier to read.

Reply:
https://github.com/GenericMappingTools/gmt/blob/8f11a6f2ebd665f825e50564e8286433be439f9f/src/gmt_init.c#L1168

This line shows that GMT always converts *xshift* to inches.